CMakeLists.txt

```diff
@@ -1,19 +1,56 @@
-cmake_minimum_required(VERSION 3.5)
-
-#-----------------------------------------------------------------------------
-# Enable C++11
-#-----------------------------------------------------------------------------
-set(_msg "Setting C++ standard")
-message(STATUS "${_msg}")
-if(NOT DEFINED CMAKE_CXX_STANDARD)
-  if("${Slicer_VTK_VERSION_MAJOR}" STREQUAL "8" OR DEFINED Qt5_DIR)
-    set(CMAKE_CXX_STANDARD 11)
-  else()
-    set(CMAKE_CXX_STANDARD 98)
+# ==== Define cmake build policies that affect compilation and linkage default behaviors
+#
+# Set the SLICER_NEWEST_VALIDATED_POLICIES_VERSION string to the newest cmake version
+# policies that provide successful builds. By setting SLICER_NEWEST_VALIDATED_POLICIES_VERSION
+# to a value greater than the oldest policies, all policies between
+# SLICER_OLDEST_VALIDATED_POLICIES_VERSION and CMAKE_VERSION (used for this build)
+# are set to their NEW behaivor, thereby suppressing policy warnings related to policies
+# between the SLICER_OLDEST_VALIDATED_POLICIES_VERSION and CMAKE_VERSION.
+#
+# CMake versions greater than the SLICER_NEWEST_VALIDATED_POLICIES_VERSION policies will
+# continue to generate policy warnings "CMake Warning (dev)...Policy CMP0XXX is not set:"
+#
+set(SLICER_OLDEST_VALIDATED_POLICIES_VERSION "3.5")
+set(SLICER_NEWEST_VALIDATED_POLICIES_VERSION "3.13.1")
+cmake_minimum_required(VERSION ${SLICER_OLDEST_VALIDATED_POLICIES_VERSION} FATAL_ERROR)
+if("${CMAKE_VERSION}" VERSION_LESS_EQUAL "${SLICER_NEWEST_VALIDATED_POLICIES_VERSION}")
+  #Set and use the newest available cmake policies that are validated to work
+  set(SLICER_CMAKE_POLICY_VERSION "${CMAKE_VERSION}")
+else()
+  set(SLICER_CMAKE_POLICY_VERSION "${SLICER_NEWEST_VALIDATED_POLICIES_VERSION}")
+endif()
+cmake_policy(VERSION ${SLICER_CMAKE_POLICY_VERSION})
+#
+# Now enumerate specific policies newer than SLICER_NEWEST_VALIDATED_POLICIES_VERSION
+# that may need to be individually set to NEW/OLD
+#
+foreach(pnew "") # Currently Empty
+  if(POLICY ${pnew})
+    cmake_policy(SET ${pnew} NEW)
+  endif()
+endforeach()
+foreach(pold "") # Currently Empty
+  if(POLICY ${pold})
+    cmake_policy(SET ${pold} OLD)
   endif()
+endforeach()
+
+# ==== Define language standard configurations requiring at least c++11 standard
+if(CMAKE_CXX_STANDARD EQUAL "98" )
+   message(FATAL_ERROR "CMAKE_CXX_STANDARD:STRING=98 is not supported in Slicer 4.11 and greater.")
```

Comment:
Since we will most likely cut a patch release of master, I would prefer we wait before integrating this.

Reply:
@jcfr, OK, but how long until this patch release is made?  If it is a day or two, that is no problem.  If the next patch release is February, I can't wait that long.   Can we start a new development branch for preparing for the next phase.  In addition to the patches currently in PR's, I also have many other patches that are not independent.  

I think i'll start a new PR "WIP:  slicer-next-phase, and just stack up changes there".

Reply:
@jcfr 

![image](https://user-images.githubusercontent.com/313970/51407783-72c71b00-1b22-11e9-9264-03ee4bd12716.png)

Are we ready now to integrate this patch?

Reply:
Yes, we will get started with supporting only to C++>=11/Qt5/VTK8 next week (if not earlier). I have a pressing deadline for tonight.